dataRetriever/blockchain/baseBlockchain.go

```diff
@@ -103,19 +102,18 @@ func (bbc *baseBlockChain) GetFinalBlockInfo() (uint64, []byte, []byte) {
 	return nonce, hash, rootHash
 }
 
-// SetCurrentAggregatedSignatureAndBitmap sets the current aggregated signature and its validator's public keys bitmap
-func (bbc *baseBlockChain) SetCurrentAggregatedSignatureAndBitmap(signature []byte, pubKeysBitmap []byte) {
+// SetCurrentHeaderProof sets the current aggregated signature and its validator's public keys bitmap
+func (bbc *baseBlockChain) SetCurrentHeaderProof(proof data.HeaderProof) {
 	bbc.mut.Lock()
 	defer bbc.mut.Unlock()
 
-	bbc.currentAggregatedSignature = signature
-	bbc.currentPubKeysBitmap = pubKeysBitmap
+	bbc.currentHeaderProof = proof
```

Comment:
this will trigger a data copy but the slices pointers will be the same. This means I can easily change here and there the slice contents after I call SetCurrentHeaderProof.
Wondering if it is better to do slice copies on each and every byte slice we provide to this function? cc @AdoAdoAdo 
As a separate task, of course.